### ISSUE TITLE
Skip statement is ignored if file is imported twice from different files

### ISSUE THREAD

**Issue by sshane** [pytest-dev/pytest#11552]
For example:

```python
# tests/common.py
import unittest


class TestBase(unittest.TestCase):
  SET_ME = False

  @classmethod
  def setUpClass(cls):
    # you can even remove the check
    if cls.__name__ == 'TestBase':
      raise unittest.SkipTest

  def setUp(self):
    # raise unittest.SkipTest  # <- this successfully skips the test
    pass

  def test_function(self):
    # this function is common but requires some class var to be set in child
    self.assertTrue(self.SET_ME)
```

And we have these two files:

```python
# tests/test_1.py
import unittest
from tests.common import TestBase

# you can even comment this class out
class Test1(TestBase):
  SET_ME = True
```

```python
# tests/test_2.py
import unittest
from tests.common import TestBase

# you can even comment this class out
class Test2(TestBase):
  SET_ME = True
```

Then run `pytest .` in `tests/` and the output is:

```python
/tests$ pytest .
==================================== test session starts ====================================
platform linux -- Python 3.11.4, pytest-7.4.3, pluggy-1.3.0
configfile: pyproject.toml
plugins: timeouts-1.2.1, subtests-0.11.0, forked-1.6.0, cpp-2.4.0, dash-2.11.1, repeat-0.9.1, timeout-2.1.0, xdist-3.3.1, anyio-3.7.1, hypothesis-6.46.7, cov-4.1.0
setup timeout: 0.0s, execution timeout: 0.0s, teardown timeout: 0.0s
collected 3 items                                                                           

test_1.py s                                                                           [ 33%]
test_2.py F.                                                                          [100%]

========================================= FAILURES ==========================================
__________________________________ TestBase.test_function ___________________________________

self = <tests.common.TestBase testMethod=test_function>

    def test_function(self):
      # this function is common but requires some class var to be set in child
>     self.assertTrue(self.SET_ME)
E     AssertionError: False is not true

common.py:19: AssertionError
=================================== slowest 10 durations ====================================

(8 durations < 0.005s hidden.  Use -vv to show these durations.)
================================== short test summary info ==================================
FAILED test_2.py::TestBase::test_function - AssertionError: False is not true
========================== 1 failed, 1 passed, 1 skipped in 0.12s ===========================
```

### Expected behavior:

```python
/tests$ python -m unittest discover .
s.s
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (skipped=2)
```

**Comment by HolyMagician03-UMich** [comment]
Hi! I can try working on this.

**Comment by HolyMagician03-UMich** [comment]
Hi! I think this issue is fixed in pytest version 8.1.1

**Comment by sshane** [comment]
Actually fixed in 8.1.0 as well. Thanks for letting me know!
